src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -519,44 +525,35 @@ public static Task<byte[]> ReadAllBytesAsync(string path, CancellationToken canc
                 return Task.FromCanceled<byte[]>(cancellationToken);
             }
 
-            var fs = new FileStream(
-                path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, // bufferSize == 1 used to avoid unnecessary buffer in FileStream
-                FileOptions.Asynchronous | FileOptions.SequentialScan);
+            // SequentialScan is a perf hint that requires extra sys-call on non-Windows OSes.
+            FileOptions options = (OperatingSystem.IsWindows() ? FileOptions.SequentialScan : FileOptions.None) | FileOptions.Asynchronous;
```

Comment:
Maybe:
```suggestion
            FileOptions options = FileOptions.Asynchronous;
            if (OperatingSystem.IsWindows())
            {
                // SequentialScan is a perf hint that requires extra sys-call on non-Windows OSes.
                options |= FileOptions.SequentialScan;
            }
```
?

Or as this is corelib, it could also be e.g.
```suggestion
            const FileOptions Options =
#if TARGET_WINDOWS
                FileOptions.Asynchronous | FileOptions.SequentialScan;
#else
                // SequentialScan is a perf hint that requires extra sys-call on non-Windows OSes.
                FileOptions.Asynchronous;
#endif                
```

Reply:
I am not convinced, as the proposals require too much space. I've moved `FileOptions.Asynchronous` to the left, hope it helps:

```diff
- FileOptions options = (OperatingSystem.IsWindows() ? FileOptions.SequentialScan : FileOptions.None) | FileOptions.Asynchronous;
+ FileOptions options = FileOptions.Asynchronous | (OperatingSystem.IsWindows() ? FileOptions.SequentialScan : FileOptions.None);
```